microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -0,0 +1,120 @@
+
+:::image type="complex" source="./images/analyze-detached-elements.msft.png" alt-text="Analyze Detached Elements in the Detached Elements tool." lightbox="./images/analyze-detached-elements.msft.png":::
+   Analyze Detached Elements in the Detached Elements tool
+:::image-end:::
+
+Complete the following steps:
+
+1. In the **Detached Elements** tool, click **Analyze**.
+
+The **Memory** tool opens in the bottom panel of DevTools.
+
+1. From the **Detached Elements** tool, select the **ID** field of a detached element.
+
+The **Memory** tool in the bottom panel automatically selects the object in the heap that is referencing the detached element. We call these objects **Retainers**.
+
+:::image type="complex" source="./images/detached-elements-heap-snapshot.msft.png" alt-text="Referencing a heap snapshot from the Detached Elements tool." lightbox="./images/detached-elements-heap-snapshot.msft.png":::
+   Referencing a heap snapshot from the Detached Elements tool.
+:::image-end:::
+
+1. From the **Memory** tool in the bottom panel, select the link **room.js:13**.
+
+The **Sources** tool opens in the top panel to line 13 of the file **room.js**.
+
+1. In the `hide()` function of **room.js**, you can see that the JavaScript code of the demo application adds each message in the room to an `unmounted` array. The `unmounted` array is the object that is referencing the detached element.
+
+:::image type="complex" source="./images/detached-elements-javascript-retainers.msft.png" alt-text="Identifying the JavaScript that is retaining the detached element." lightbox="./images/detached-elements-javascript-retainers.msft.png":::
+   Identifying the JavaScript that is retaining the detached element.
+:::image-end:::
+
+You have now identified the retainer that is preventing the detached element from being garbage collected by the browser!
+
+## Next steps
+
+When looking for memory leaks, remember that leaks can depend on the context of your application. For the demo application, you found detached elements that could not be garbage collected by the browser and identified the JavaScript retaining the detached elements. However, in the context of the demo application, it makes sense to retain the list of chat messages so that if a user switches back to **Room 1**, the message log is preserved.
+
+:::image type="complex" source="./images/detached-elements-reattached.msft.png" alt-text="The detached elements are reattached to the DOM when switching back to Room 1." lightbox="./images/detached-elements-reattached.msft.png":::
+   The detached elements are reattached to the DOM when switching back to Room 1.
+:::image-end:::
```

Comment:
It's not entirely clear to me what this image illustrates without the description (which is only visible if I click on the image to see the lightbox).
The text before it makes sense to me, and is enough for me to understand the point. I don't think the image really adds much value. If we want to keep it, we should add a clear description above it, in the article.